### DISCUSSION TITLE
Is there a migration guide?

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#1196] [General]
Hello! It's been a little while since I've been here and you all seem to be moving fast! Do you have a migration guide for each release? In particular, it would be great to know what to update on an already-bootstrapped project to get to the latest Langium.

I could bootstrap two projects, one at the old version, one at the new version, and use a diff tool to see the differences, but that wouldn't be as ideal as something like the migration guide that Three.js has:

https://github.com/mrdoob/three.js/wiki/Migration-Guide

Every release has the exact details on how to migrate to the next version.

**Comment by montymxb** [comment]
Hey @trusktr . You might find the [changelogs](https://github.com/eclipse-langium/langium/blob/main/CHANGELOG.md) helpful to address version-by-version changes. Although not a de-factor migration guide, it does give a nice overview of the changes you should generally expect across versions. In particular, the [Langium changelog](https://github.com/eclipse-langium/langium/blob/main/packages/langium/CHANGELOG.md) has details on the recent change to ESM.

**Reply by trusktr** [reply]
Ah thanks. I missed the nested changelog files. I'll look at those when updating, and come back if needed.
